components/hydrator/internal/certsubjectmapping/loader.go

```diff
@@ -64,35 +71,72 @@ func (s *SubjectConsumerTypeMapping) Validate() error {
 	return nil
 }
 
-func StartCertSubjectMappingLoader(ctx context.Context, certSubjectMappingCfg Config, directorClient DirectorClient) Cache {
+func StartCertSubjectMappingLoader(ctx context.Context, certSubjectMappingCfg Config, directorClient DirectorClient) (Cache, error) {
 	cache := NewCertSubjectMappingCache()
 	certSubjectLoader := NewCertSubjectMappingLoader(cache, certSubjectMappingCfg, directorClient)
+
+	err := certSubjectLoader.InitialiseCertSubjectMappings(ctx, certSubjectMappingCfg.EnvironmentMappings)
+	if err != nil {
+		return nil, err
+	}
+
 	go certSubjectLoader.Run(ctx, certSubjectMappingCfg.EnvironmentMappings)
 
-	return cache
+	return cache, nil
 }
 
-func (cl *certSubjectMappingLoader) Run(ctx context.Context, certSubjectMappingsFromEnv string) {
+func (cl *certSubjectMappingLoader) InitialiseCertSubjectMappings(ctx context.Context, certSubjectMappingsFromEnv string) error {
 	entry := log.C(ctx)
 	entry = entry.WithField(log.FieldRequestID, CertSubjectMappingLoaderCorrelationID)
 	ctx = log.ContextWithLogger(ctx, entry)
 
-	t := time.NewTicker(cl.certSubjectMappingCfg.ResyncInterval)
-	for {
-		mappings, err := cl.loadCertSubjectMappings(ctx, certSubjectMappingsFromEnv)
+	var mappings []SubjectConsumerTypeMapping
+	var err error
+
+	err = retry.Do(func() error {
+		mappings, err = cl.loadCertSubjectMappings(ctx, certSubjectMappingsFromEnv)
 		if err != nil {
-			log.C(ctx).WithError(err).Errorf("Certificate subject mapping resync failed with error: %v", err)
-		} else {
-			log.C(ctx).Info("Update certificate subject mapping cache with the newly fetched data")
-			cl.certSubjectMappingCache.Put(mappings)
+			return err
 		}
+		return nil
+	},
+		retry.Attempts(0), // we want the default value here
```

Comment:
Bad wording from my side.

We want it to be 0 so that the retries don't have upper bound. If it succeeds, we are good and continue. If it keeps failing and failing, the pod will be stuck and won't be initialised and we leave the decision to k8s when to terminate it and schedule a new one.